### PR TITLE
[ISSUE #143] add integration test for restlight

### LINKED ISSUE #143: [Integration Test] add integration test for springmvc.
### Expected behavior
add integration test for restlight.

### Actual behavior
have no integration test.

### Steps to reproduce
none

### Env
- Restlight version: 1.0.0-SNAPSHOT
- JVM version: 1.8
- OS:Windows 10

### PR DESCRIPTION
**Pull request by brotherlu-xcq**
### Describe what this PR does / why we need it
add integration test for restlight.

### Describe how you did it

Fixes #143 

### Describe how to verify it
check and run the workflow

### Special notes for reviews
no

### CODE REVIEW

**Comment by alalag1** [comment]
LGTM

File: restlight-integration-test/springmvc-test/src/main/java/io/esastack/restlight/integration/cases/controller/FailFastController.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.concurrent.Executor;
+
+/**
+ * @author chenglu
+ */
+@RestController
+@RequestMapping("/failfast/")
+public class FailFastController implements RestlightBizExecutorAware {
```
Comment (LCDZhao-Z) [inline restlight-integration-test/springmvc-test/src/main/java/io/esastack/restlight/integration/cases/controller/FailFastController.java:30]: Is it necessary to implement RestlightBizExecutorAware？

File: restlight-integration-test/springmvc-test/src/main/java/io/esastack/restlight/integration/cases/controller/ExceptionController.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+package io.esastack.restlight.integration.cases.controller;
+
+import io.esastack.restlight.integration.cases.exception.CustomException;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+/**
+ * @author chenglu
```
Comment (alalag1) [inline restlight-integration-test/springmvc-test/src/main/java/io/esastack/restlight/integration/cases/controller/ExceptionController.java:22]: do not import your name plz

Comment (alalag1) [reply restlight-integration-test/springmvc-test/src/main/java/io/esastack/restlight/integration/cases/controller/ExceptionController.java:22]: same with other classes. check plz

**Review by alalag1 — CHANGES_REQUESTED** [review]

File: restlight-integration-test/springmvc-test/src/main/java/io/esastack/restlight/integration/cases/controller/FilterController.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+/**
+ * @author chenglu
+ */
+@RestController
+@RequestMapping("/filter/")
+public class FilterController {
+
+    @GetMapping("get")
+    public UserData filter(@RequestHeader String name) {
+        return UserData.Builder.aRestResult()
```
Comment (LCDZhao-Z) [inline restlight-integration-test/springmvc-test/src/main/java/io/esastack/restlight/integration/cases/controller/FilterController.java:31]: Does aRestResult() have any special meaning? Logically, it is not necessary to call this method to build UserData.

File: restlight-integration-test/springmvc-test/src/main/java/io/esastack/restlight/integration/cases/inteceptor/AttributeInterceptor.java [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+import esa.commons.collection.AttributeKey;
+import io.esastack.restlight.core.interceptor.InternalInterceptor;
+import io.esastack.restlight.server.context.RequestContext;
+
+import java.util.concurrent.CompletionStage;
+
+/**
+ * @author chenglu
+ */
+public class AttributeInterceptor implements InternalInterceptor {
```
Comment (LCDZhao-Z) [inline restlight-integration-test/springmvc-test/src/main/java/io/esastack/restlight/integration/cases/inteceptor/AttributeInterceptor.java:25]: Is it necessary to set a separate package for AttributeInterceptor  when the package is not set independently for filters and some paramResolvers?

File: restlight-integration-test/springmvc-test/src/main/java/io/esastack/restlight/integration/cases/resolver/GlobalExceptionResolver.java [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+import org.springframework.stereotype.Component;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+
+/**
+ * @author chenglu
+ */
+@Component
+public class GlobalExceptionResolver implements ExceptionResolver<IllegalArgumentException> {
```
Comment (LCDZhao-Z) [inline restlight-integration-test/springmvc-test/src/main/java/io/esastack/restlight/integration/cases/resolver/GlobalExceptionResolver.java:28]: Is it necessary to set a separate package for GlobalExceptionResolver  when the package is not set independently for filters and some paramResolvers?

File: restlight-integration-test/springmvc-test/src/main/java/io/esastack/restlight/integration/cases/resolver/CustomExceptionHandler.java [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+import io.esastack.restlight.integration.cases.exception.CustomException;
+import io.esastack.restlight.server.core.HttpResponse;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+
+/**
+ * @author chenglu
+ */
+//@ControllerAdvice
+public class CustomExceptionHandler {
```
Comment (LCDZhao-Z) [inline restlight-integration-test/springmvc-test/src/main/java/io/esastack/restlight/integration/cases/resolver/CustomExceptionHandler.java:26]: How does it work?

File: restlight-integration-test/springmvc-test/src/main/java/io/esastack/restlight/integration/cases/config/SerializeConfig.java [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+    }
+
+    @Bean
+    public HttpBodySerializer gsonHttpBodySerializer() {
+        return new GsonHttpBodySerializer();
+    }
+
+    @Bean
+    public HttpBodySerializer jacksonHttpBodySerializer() {
+        return new JacksonHttpBodySerializer() {
```
Comment (LCDZhao-Z) [inline restlight-integration-test/springmvc-test/src/main/java/io/esastack/restlight/integration/cases/config/SerializeConfig.java:42]: Is it needed to inject JacksonHttpBodySerializer when it had been loaded by spi automatically in Deployments.loadDefaultSerializersIfNecessary() ?

**Pull request merged**